src/components/AddAssignment.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import Cookies from 'js-cookie';
+import { SERVER_URL } from '../constants.js';
+
+class AddAssignment extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { name: '', dueDate: '', course: '', redirect: false };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    event.preventDefault();
+    switch (event.target.id) {
+      case 'name':
+        this.setState({ name: event.target.value });
+        break;
```

Comment:
Maybe instead of break, you can put like " "

Reply:
I think break is required for a switch case:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch